### PR TITLE
Fix Bitop Edge Case

### PR DESCRIPTION
**Pull request by vazois**
This PR fixes edge case where bitop will result in null ptr exception when none of the source keys exist.
I added also some more negative tests for bitop.

### CODE REVIEW

**Comment by badrishc** [comment]
cc @kevin-montrose

File: libs/server/Resp/Bitmap/BitmapCommands.cs [hunk 1]
```diff
@@ -450,19 +459,34 @@ private bool StringBitOperation<TGarnetApi>(int count, byte* ptr, BitmapOperatio
...
             }
 
-            var status = storageApi.StringBitOperation(keys, bitop, out long result);
-
-            if (status != GarnetStatus.NOTFOUND)
+            if (sizeof(byte*) * (keyCount - 1) > 512)
             {
-                while (!RespWriteUtils.WriteInteger(result, ref dcurr, dend))
-                    SendAndReset();
+                throw new Exception("Bitop source key limit (64) exceeded");
```
Comment (badrishc) [inline libs/server/Resp/Bitmap/BitmapCommands.cs:483]: What does this exception surface to client code as?

File: libs/server/Resp/RespCommand.cs [hunk 2]
```diff
@@ -460,47 +460,7 @@ static RespCommand MatchedNone(RespServerSession session, int oldReadHead)
...
-                                                            return (RespCommand.BITOP, (byte)BitmapOperation.NOT);
-                                                        }
-
-                                                        // Reset read head and count if we didn't match operator.
-                                                        readHead -= 9;
-                                                        count += 1;
-                                                    }
-                                                }
-                                            }
+                                            return (RespCommand.BITOP, 0);
```
Comment (badrishc) [inline libs/server/Resp/RespCommand.cs:463]: Why was this change reverted? Our goal going forward is to try and consolidate all the command and subcommand parsing into the parsing phase, rather than depend on individual operations to parse. That way ACL can also be managed centrally after the (command, subcommand) are known.

File: libs/server/Resp/Bitmap/BitmapCommands.cs (hunk 1 above)
Comment (vazois) [reply libs/server/Resp/Bitmap/BitmapCommands.cs]: Good point. I added a RESP response that is surfaced to the client instead of throwing an exception.

File: libs/server/Resp/RespCommand.cs (hunk 2 above)
Comment (vazois) [reply libs/server/Resp/RespCommand.cs]: Whereas the most data commands I have seen follow the format <command> <key>, BITOP expects the key after the operation type (i.e. AND,OR, XOR etc.) which could be considered as subcommand, though technically it isn't.
I made the change in order to deal with the event the provided operation type is not valid.
I reverted it back to the original state and added a workaround.
Now if BITOP is matched and then the subcommand is not matched we still end up in BITOP but capture the error using BitmapOperation.NONE flag and drain the buffer.

**Review by badrishc — APPROVED** [review]
